src/send/hooks.ts

```diff
@@ -0,0 +1,226 @@
+import { useState, useMemo, useEffect } from 'react'
+import { debounce, throttle } from 'lodash'
+import useSelector from 'src/redux/useSelector'
+import { useAsync } from 'react-async-hook'
+import { phoneRecipientCacheSelector } from 'src/recipients/reducer'
+import { recipientInfoSelector } from 'src/recipients/reducer'
+import { isValidAddress } from '@celo/utils/lib/address'
+import { parsePhoneNumber } from '@celo/phone-utils'
+import {
+  sortRecipients,
+  getRecipientFromAddress,
+  Recipient,
+  RecipientType,
+  filterRecipientFactory,
+} from 'src/recipients/recipient'
+import { defaultCountryCodeSelector } from 'src/account/selectors'
+import { useTranslation } from 'react-i18next'
+import { NameResolution, ResolutionKind } from '@valora/resolve-kit'
+import { resolveId } from 'src/recipients/RecipientPicker'
+
+const TYPING_DEBOUNCE_MILLSECONDS = 300
+const SEARCH_THROTTLE_TIME = 100
+
+/**
+ * Returns a single ordered list of all recipients to show in search results,
+ * as well as the search query state variable itself and its setter.
+ *
+ * onSearch is a callback function which will be called with the search query
+ * just before it's updated.
+ *
+ * This hook is tested via the SendSelectRecipient.test.tsx file.
+ */
+export function useMergedSearchRecipients(onSearch: (searchQuery: string) => void) {
```

Comment:
I think something strange is happing in this function. Currently I only get one result regardless of my query. For example if I search an area code I only see one result for that area code despite having several contacts with matching area codes.